HADotNet.Core/BaseClient.cs

```diff
@@ -24,6 +25,7 @@ public abstract class BaseClient
         protected BaseClient(Uri instance, string apiKey)
         {
             Client = new RestClient(instance);
+            Client.UseNewtonsoftJson();
```

Comment:
Since we are manually calling `JsonConvert.DeserializeObject()`, we don't need to implement this. We are only using RestSharp for transport, not for (de)serialization.

Reply:
Jake, I got your point, and resolved my issues by explicitly serializing objects inside my Client class before calling `BaseClient.Post` .